packages/client/src/ActionsAndReducers/playerUi/helpers/handleWargameMessagesChange.ts

```diff
@@ -30,233 +27,24 @@ import {
-  expiredStorage.getItem(`${currentWargame}-${selectedForce}-${selectedRole}${id}`) === 'read'
-)
-
-export const handleSetAllMEssages = (payload: (MessageInfoType | MessageCustom)[], newState: PlayerUi): SetWargameMessage => {
-
-  const messagesFiltered: Array<MessageChannel> = payload.map((message) => {
-    if (message.messageType === INFO_MESSAGE) {
-      const res: MessageInfoTypeClipped = {
-        messageType: INFO_MESSAGE_CLIPPED,
-        details: {
-          channel: `infoTypeChannelMarker${uniqId.time()}`
-        },
-        infoType: true,
-        gameTurn: message.gameTurn,
-        isOpen: false,
-        hasBeenRead: typeof message._id === 'string' && isMessageHasBeenRead(message._id, newState),
-        _id: message._id
-      }
-
-      return res
-    }
-
-    return {
-      ...message,
-      hasBeenRead: isMessageHasBeenRead(message._id, newState),
-      isOpen: false
-    }
-  })
-
-  const messages = _.uniqBy(messagesFiltered, reduceTurnMarkers)
-    .filter((message) => message.details && message.details.channel === newState.chatChannel.name)
-
-  const channels = {}
-
-  newState.allChannels.forEach((channel) => {
-    const {
-      isParticipant,
-      allRolesIncluded,
-      observing,
-      templates
-    } = getParticipantStates(channel, newState)
-
-    if (!newState.isObserver && !isParticipant && !allRolesIncluded) return
-    else {
-      // TODO: use channel name
-      channels[channel.uniqid || channel.name] = {
-        name: channel.name,
-        templates,
-        forceIcons: channel.participants && channel.participants.map((participant) => participant.icon),
-        forceColors: channel.participants && channel.participants.map((participant) => {
-          const force = newState.allForces.find((force) => force.uniqid === participant.forceUniqid)
-          return (force && force.color) || '#FFF'
-        }),
-        messages: messages.filter((message) => message.details && message.details.channel === channel.uniqid || message.messageType === INFO_MESSAGE_CLIPPED),
-        unreadMessageCount: messages.filter((message) => {
-          if (message.messageType !== INFO_MESSAGE_CLIPPED) {
-            return false
-          } else {
-            return (
-              expiredStorage.getItem(`${newState.currentWargame}-${newState.selectedForce}-${newState.selectedRole}${message._id}`) === null &&
-              message.details.channel === channel.uniqid
-            )
-          }
-        }).length,
-        observing
-      }
-    }
-  })
-
-  return {
-    channels,
-    chatChannel: {
-      ...newState.chatChannel,
-      messages
-    }
-  }
+/** when the app first opens it processes a list of all existing messages,,
+ * grouping them into channels
+ */
+export const handleSetAllMessages = (payload: Array<MessageCustom | MessageInfoType>, newState: PlayerUi): SetWargameMessage => {
```

Comment:
@lilitkarapetyan  - it felt great getting these blocks of code into testable helpers.

I really enjoyed working in the `helpers` package.

I was able to run `yarn test:watch handle` - and each time I pressed `Save` it ran through the matching test(s) again.  It's also much quicker development cycle than doing a full build & driving the UI to test bits.